biggerquery/build.py

```diff
@@ -0,0 +1,292 @@
+    return docker_repository
+
+
+def secure_get_version():
+    try:
+        return get_version()
+    except Exception as e:
+        print(e)
+        raise ValueError("Can't get the current package version. To use the automatic versioning, "
+                         "you need to use git inside your project directory.")
+
+
+def auto_configuration(project_name: str, project_dir: Path = Path('.').parent):
+    '''
+    Auto configuration for the standard BigFlow project structure (that you can generate through the CLI).
+    The 'project_name' parameter should be a valid python package name.
+
+    Example:
+    project_setup(**auto_configuration('my_super_project'))
+    '''
+    deployment_config_file = project_dir / 'deployment_config.py'
+
+    return {
+        'project_name': project_name,
+        'docker_repository': get_docker_repository_from_deployment_config(deployment_config_file),
+        'root_package': project_dir / project_name,
+        'project_dir': project_dir,
+        'build_dir': project_dir / 'build',
+        'test_package': project_dir / 'test',
+        'dags_dir': project_dir / '.dags',
+        'dist_dir': project_dir / 'dist',
+        'image_dir': project_dir / 'image',
+        'eggs_dir': project_dir / f'{project_name}.egg-info',
+        'deployment_config_file': deployment_config_file,
+        'version': secure_get_version(),
+        'resources_dir': project_dir / 'resources',
+        'project_requirements_file': project_dir / 'resources' / 'requirements.txt'
+    }
+
+
+def project_setup(
+        project_name: str,
+        docker_repository: str,
+        root_package: Path,
+        project_dir: Path,
+        build_dir: Path,
+        test_package: Path,
+        dags_dir: Path,
+        dist_dir: Path,
+        image_dir: Path,
+        eggs_dir: Path,
+        deployment_config_file: Path,
+        version: str,
+        resources_dir: Path,
+        project_requirements_file: Path):
+    '''
+    This function produces arguments for setuptools.setup. The produced setup provides commands that allow you to build
+    whl package, docker image and DAGs. Paired with auto_configuration function, it provides fully automated build
+    tool (accessed by CLI) for your project (that you can generate using CLI).
+
+    Example:
+    from setuptools import setup
+    from biggerquery.build import project_setup, auto_configuration
+
+    setup(project_setup(**auto_configuration('my_super_project')))
+    '''
+    if project_name is None or docker_repository is None or root_package is None or project_dir is None or \
+            build_dir is None or test_package is None or dags_dir is None or dist_dir is None or image_dir is None or \
+            eggs_dir is None or deployment_config_file is None or version is None or resources_dir is None or project_requirements_file is None:
+        raise ValueError('You need to provide all parameters for the project_setup function.')
```

Comment:
Could we provide explictly info about missing parameters?

Reply:
done